presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestUnion.java

```diff
@@ -63,6 +64,7 @@ public void testPartialRollupAggregationsWithUnion()
                 LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED,
                 false);
         assertAtMostOneAggregationBetweenRemoteExchanges(plan);
+        assertPlanIsFullyDistributed(plan);
     }
```

Comment:
Also, is there an explicit test for source partitioned + UNION + group by? Or something else that results in non-perfect partitioning?